python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -1835,16 +1841,39 @@ def run(self, ccdExposure, *, dnlLUT=None, bias=None, deferredChargeCalib=None,
                 maskPlaneNames=self.config.itlDipMaskPlanes,
             )
 
-        # Edge bleed masking
-        if self.config.doITLEdgeBleedMask and detector.getPhysicalType() == 'ITL' \
-                and self.config.doSaturation:
-            isrFunctions.maskITLEdgeBleed(ccdExposure=ccdExposure,
-                                          itlEdgeBleedSatMinArea=self.config.itlEdgeBleedSatMinArea,
-                                          itlEdgeBleedSatMaxArea=self.config.itlEdgeBleedSatMaxArea,
-                                          itlEdgeBleedThreshold=self.config.itlEdgeBleedThreshold,
-                                          itlEdgeBleedModelConstant=self.config.itlEdgeBleedModelConstant,
-                                          saturatedMaskName=self.config.saturatedMaskName,
-                                          badAmpDict=badAmpDict)
+        if (self.config.doITLSatSagMask or self.config.doITLEdgeBleedMask) \
+                and detector.getPhysicalType() == 'ITL':
```

Comment:
This is much more logic than should be in the parent run method; why isn't this a function?  (Either a self.method or an isrFunction; an isrFunction is more testable)

Reply:
Because of the possibility of two cores in a footprint, these masking now work on footprints rather than on the exposure but I only want trigger this loop if the masking configs are true.

Reply:
I mean below this is way too much logic for the core run method of `IsrTaskLSST`, and should be moved either into a method or an isr function.

Reply:
I'd vote method, with the unit test being just a direct call to that method with a faked footprint that needs to be split.